### DISCUSSION TITLE
Adding format: revealjs YAML front matter using a filter, good or bad idea?

### DISCUSSION THREAD

**Discussion by cavo789** [quarto-dev/quarto-cli#8219] [Q&A] [answered]
### Description

Hi all

I'm busy to convert my notes (a lot of) from RevealJS to Quarto i.e. one single document with multiple output formats.

In short, I've a lot of documents having something like:

```markdown
---
title: "My Slide Show"
subtitle: "Playing with logo"
author: "Christophe"
date: last-modified
lang: "en"
format:
  revealjs:
    logo: images/my-logo.svg
    css: assets/style.css
    resources: images
    header-includes: |
      <script src="assets/custom.js" type="application/javascript"></script>
    << and a lot more properties >>
---
```

In my YAML, I've thus a `format -> revealjs` entry and a lot of settings.

Right now, I've five slideshows and I need to migrate somes more.

My idea was : stop to put `format -> revealjs` in each `slides.md` files but create a small filter `add_reveal_settings` and that one will inject the `format -> revealjs` settings.

So, if my idea is correct, I can just replace my front matter with:

```markdown
---
title: "My Slide Show"
subtitle: "Playing with logo"
author: "Christophe"
date: last-modified
lang: "en"
filter:
   - add_reveal_settings
---
```

and get the exact same things (since `add_reveal_settings` will add every single settings).

Is this a good idea? Should I need to create a filter for this (did you know if I can reuse an existing filter; which one?)
Can I do the same thing native?

Many thanks to the community.

**Comment by cscheid** [comment]
That's unfortunately not going to work; `format` is a key that needs to be available _before_ we invoke Pandoc.

**Reply by cavo789** [reply]
Voilà! So I know I should no more investigate that direction.

How did you proceed? Are you copy/pasting your format -> revealjs settings in every single slideshow?  And when you discover a nice new setting (like today for me with `slide-level`; edit all slideshows and add the setting manually) ? 

Thanks

**Comment by cderv** [comment, marked as answer]
Are your presentations in the same project ? 

If so, then you can have shared metadata through `_quarto.yml` or `_metadata.yml`. Doc is at https://quarto.org/docs/projects/quarto-projects.html#shared-metadata

Or are your presentation in different projects ? 

In that case, the way to share a format accross project would be to package it as an extension to create a custom format that is just a customed version of existing `revealjs` format. 

More on that at https://quarto.org/docs/extensions/formats.html

It needs to be included in each presentation project aftet, but you can then distribute your extension several way and easily install it / update with quarto CLI

Hope it helps

**Reply by cavo789** [reply]
Thank you Christophe.

Different projects indeed so, yes, probably should I need to define my own template as an extension. Thanks for pointing the way to go.

**Comment by cavo789** [comment]
I've made additionals tests and, yes, it's not so difficult to create his own revealjs extension.

I've mainly look at extensions from Michaël like https://github.com/mcanouil/quarto-revealjs-coeos.

Thanks guys!
